### PR TITLE
Set OS_LIBS under Windows

### PR DESCRIPTION
**Pull request by mgondan**
This mimics the CMakeLists from package ssl. Relevant for builds with static libarchive (rtools42).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -64,7 +67,7 @@ swipl_plugin(
     archive
     MODULE archive4pl
     C_SOURCES archive4pl.c
-    C_LIBS ${LibArchive_LIBRARIES}
+    C_LIBS ${LibArchive_LIBRARIES} ${LibArchive_OS_LIBS}
```
Comment (JanWielemaker) [inline CMakeLists.txt:70]: I think the OS libs are already included above, no?

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,16 +7,19 @@ if(POLICY CMP0074)
...
-# the libarchive dependencies
-if(LibArchive_LIBRARIES AND LibArchive_INCLUDE_DIRS)
-  set(LibArchive_FOUND ON)
+find_package(LibArchive)
+
+if(LibArchive_FOUND)
+
+if(NOT LibArchive_OS_LIBS)
+if(WIN32)
+  set(LibArchive_OS_LIBS nettle.lib xml2.lib iconv.lib regex.lib expat.lib lzma.lib zstd.lib lz4.lib bz2.lib z.lib bcrypt.lib ws2_32.lib)
```
Comment (JanWielemaker) [inline CMakeLists.txt:16]: Hmm.  This is not needed (and undesirable) for the DLL version.   Another problem is that the set of libraries depends on the included formats and filters in libarchive.    I fear we need a more dynamic way to get this list 😢

**Comment by mgondan** [comment]
It was needed in both places,

- above, to get the AC-checks to work
- here, to make the module link (see the corresponding place in package ssl)

Best wishes,

Matthias

Am 25.04.2022 um 10:19 schrieb Jan Wielemaker ***@***.******@***.***>>:


@JanWielemaker commented on this pull request.

________________________________

In CMakeLists.txt<https://github.com/SWI-Prolog/packages-archive/pull/20#discussion_r857370830>:

> @@ -64,7 +67,7 @@ swipl_plugin(
     archive
     MODULE archive4pl
     C_SOURCES archive4pl.c
-    C_LIBS ${LibArchive_LIBRARIES}
+    C_LIBS ${LibArchive_LIBRARIES} ${LibArchive_OS_LIBS}


I think the OS libs are already included above, no?

**Comment by mgondan** [comment]
Ok, l'll come back with a less intrusive PR that just allows a switch with the "OS libs"

Matthias

Am 25.04.2022 um 10:21 schrieb Jan Wielemaker ***@***.******@***.***>>:


@JanWielemaker commented on this pull request.

________________________________

In CMakeLists.txt<https://github.com/SWI-Prolog/packages-archive/pull/20#discussion_r857372629>:

> @@ -7,16 +7,19 @@ if(POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
 endif()

-# The native FindLibArchive is rather naive.  This allows overruling
-# the libarchive dependencies
-if(LibArchive_LIBRARIES AND LibArchive_INCLUDE_DIRS)
-  set(LibArchive_FOUND ON)
+find_package(LibArchive)
+
+if(LibArchive_FOUND)
+
+if(NOT LibArchive_OS_LIBS)
+if(WIN32)
+  set(LibArchive_OS_LIBS nettle.lib xml2.lib iconv.lib regex.lib expat.lib lzma.lib zstd.lib lz4.lib bz2.lib z.lib bcrypt.lib ws2_32.lib)


Hmm. This is not needed (and undesirable) for the DLL version. Another problem is that the set of libraries depends on the included formats and filters in libarchive. I fear we need a more dynamic way to get this list 😢

**Pull request closed without merging**